File: frontend/src/app/facade/Parkplatz.ts
import { AktivitaetsEnum } from "../utils/aktivitaetEnum.utils"
import { Parkplatztyp } from "./Parkplatztyp"
import { Preiskategorie } from "./Preiskategorie"

export interface Parkplatz {
    parkplatzID: number
    nummer: string
    xkoordinate: number
    ykoordinate: number
    aktivitaet: AktivitaetsEnum
    parkplatztyp: Parkplatztyp
    preiskategorie: Preiskategorie
}

File: backend/src/main/java/com/gfi/parkplatzapp/backend/persistence/entities/Parkplatz.java
package com.gfi.parkplatzapp.backend.persistence.entities;

import com.gfi.parkplatzapp.backend.utils.AktivitaetEnum;
import jakarta.persistence.*;
import lombok.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Parkplatz {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long parkplatzID;

    private String nummer;
    private int xKoordinate;
    private int yKoordinate;

    @Enumerated(EnumType.STRING)
    private AktivitaetEnum aktivitaet;
    @ManyToOne
    private Parkplatztyp parkplatztyp;
    @ManyToOne
    private Preiskategorie preiskategorie;
    @ManyToOne(cascade = CascadeType.ALL)
    private Parkflaeche parkflaeche;
}
